### PR TITLE
Support @RepositoryDefinition for Spring Data repositories

### LINKED ISSUE #10614: Quarkus extension for spring data API - automatic repository implementation generation
Hello,

I am trying to migrate a spring project in quarkus and I am using quarkus extension for spring data API.

As they said in the quarkus doc : Interfaces that extend org.springframework.data.repository.Repository are automatically implemented and the generated repositories are also registered as beans so they can be injected into any other bean (source : https://quarkus.io/guides/spring-data-jpa#what-is-supported). But in the spring doc they said that instead of extending org.springframework.data.repository.Repository we can alternatively annotate our interface with @RepositoryDefinition and it will cause the same behaviour as extending org.springframework.data.repository.Repository (source : https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.definition-tuning).

My question is : Does spring data API generate automatically the repository that is annotated with @RepositoryDefinition since it has the same behavior as extending org.springframework.data.repository.Repository ?
If no, is there any other way to make spring data API generate the repository that is annotated with @RepositoryDefinition ?

Note that I can't modify the package where the interface is declared since it is a third party library. 

Thanks in advance

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #10614

### CODE REVIEW

**Comment by geoand** [comment]
@gsmet although this isn't technically a bug fix, it is however really small and doesn't affect anything else so I would like to have it in `1.6.1.Final` if possible :)

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/SpringDataRepositoryCreator.java [hunk 1]
```diff
@@ -142,8 +143,15 @@ public void implementCrudRepository(ClassInfo repositoryToImplement) {
         }
 
         if (idDotName == null || entityDotName == null) {
-            throw new IllegalArgumentException(
-                    "Repository " + repositoryToImplement + " does not specify ID and/or Entity type");
+            AnnotationInstance repositoryDefinitionInstance = repositoryToImplement
+                    .classAnnotation(DotNames.SPRING_DATA_REPOSITORY_DEFINITION);
+            if (repositoryDefinitionInstance != null) {
+                entityDotName = repositoryDefinitionInstance.value("domainClass").asClass().name();
+                idDotName = repositoryDefinitionInstance.value("idClass").asClass().name();
```
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/SpringDataRepositoryCreator.java:150]: I'm not sure about the place of this. If `@RepositoryDefinition` is used, can't we skip all the code above?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a question, we need to clarify it before merging.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/SpringDataRepositoryCreator.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/SpringDataRepositoryCreator.java]: You are absolutely right! Fixed

**Review by gsmet — APPROVED** [review]

**Pull request merged**